superset/sql_parse.py

```diff
@@ -111,7 +111,11 @@ def is_select(self) -> bool:
         return self._parsed[0].get_type() == "SELECT"
 
     def is_explain(self) -> bool:
-        return self.stripped().upper().startswith("EXPLAIN")
+        # Parsing SQL statement for EXPLAIN and filtering out comments
```

Comment:
Thinking about whether the right logic here is `first()`, `any()` or `all()` and I think it's either 

- 1) it works only if the SQL is a single statement, and raises otherwise
- 2) all of the statements are `EXPLAIN` (which is not typical)